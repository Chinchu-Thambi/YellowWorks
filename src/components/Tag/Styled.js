import styled from 'styled-components';

export const BaseTag = styled.span`
  user-select: none;
  display: inline-flex;
  border-radius: ${({ theme }) => theme.space[1]};
  color: ${({ theme }) => theme.palette.contrast[2]}; 
  border: 1px solid  ${({ theme }) => theme.palette.contrast[2]};
  padding: ${({ theme }) => theme.space[1]} ${({ theme }) => theme.space[2]};
  margin: 0 ${({ theme }) => theme.space[1]} ${({ theme }) => theme.space[1]} 0;
  font-size: ${({ theme }) => theme.fontSizes[2]};
  line-height: ${({ theme }) => theme.space[3]};

  background-color: ${({ highlight, theme }) => (highlight ? theme.palette.navigation[4] : 'transparent')};
  transition: background-color .4s;
`;

export const RemoveTagButton = styled.button`
  border: 0;
  padding: 0;
  width: 18px;
  height: 18px;
  cursor: pointer;
  background: transparent;
  margin-left: ${({ theme }) => theme.space[1]};

  svg {
    margin-top: 1px;
    transform: rotate(45deg);

    circle {
      fill: ${({ theme }) => theme.palette.contrast[2]}; 
    }
  }

  &:hover {
    svg > circle {
      fill: ${({ theme }) => theme.palette.accent[3][0]};
    }
  }
`;
