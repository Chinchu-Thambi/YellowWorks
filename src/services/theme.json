{
  "colors": {
    "base": {
      "100": "#FFFFFF",
      "200": "#F4F4F4",
      "300": "#F2F2F2",
      "400": "#E7E7E7"
    },
    "brand": {
      "100": "#FFFDF4",
      "200": "#FEF9D3",
      "300": "#FDF3A7",
      "400": "#FDED7A",
      "500": "#FCE74E",
      "600": "#FBE122"
    },
    "contrast": {
      "100": "#F3F5F8",
      "200": "#D2D4DA",
      "300": "#A5AAB4",
      "400": "#797F8F",
      "500": "#4C5569",
      "600": "#1F2A44"
    },
    "crimson": {
      "100": "#FDF5F5",
      "200": "#F7D9D8",
      "300": "#EFB2B1",
      "400": "#E88C8B",
      "500": "#E06564",
      "600": "#D83F3D"
    },
    "fern": {
      "100": "#F7FBF6",
      "200": "#E0EFDD",
      "300": "#C1DFBB",
      "400": "#A2D09A",
      "500": "#83C078",
      "600": "#64B056"
    },
    "uplift": {
      "100": "#FEF8F6",
      "200": "#FBE2DA",
      "300": "#F8C6B5",
      "400": "#F4A991",
      "500": "#F18D6C",
      "600": "#ED7047"
    },
    "pacific": {
      "100": "#F7FBFE",
      "200": "#E1F2FA",
      "300": "#C4E5F5",
      "400": "#A6D8F1",
      "500": "#89CBEC",
      "600": "#6BBEE7"
    },
    "magnificent": {
      "100": "#F7F5FA",
      "200": "#E2D8EA",
      "300": "#C5B0D5",
      "400": "#A789C0",
      "500": "#8A61AB",
      "600": "#6D3A96"
    },
    "violet": {
      "100": "#F7F7FA",
      "200": "#DFE1EC",
      "300": "#C0C2D8",
      "400": "#A0A4C5",
      "500": "#8185B1",
      "600": "#61679E"
    }
  },
  "opacity": {
    "95": "0.95"
  },
  "spacing": {
    "px": "1px",
    "0": "0rem",
    "1": "0.3125rem",
    "2": "0.625rem",
    "2.5": "1rem",
    "3": "1.25rem",
    "4": "2rem",
    "5": "3rem",
    "6": "4rem",
    "7": "5rem",
    "8": "6rem",
    "9": "7rem",
    "10": "8rem",
    "11": "10rem",
    "12": "12rem",
    "13": "14rem",
    "14": "16rem",
    "15": "18rem",
    "16": "20rem",
    "17": "24rem",
    "18": "32rem",
    "19": "36rem",
    "20": "48rem"
  },
  "screens": {
    "sm": "36em",
    "md": "48em",
    "lg": "62em",
    "xl": "75em",
    "2xl": "90em"
  },
  "borderRadius": {
    "none": "0",
    "sm": "0.125rem",
    "default": "0.15625rem",
    "md": "0.3125rem",
    "lg": "0.625rem",
    "xl": "0.75rem",
    "2xl": "1rem",
    "3xl": "1.5rem",
    "full": "9999px"
  },
  "boxShadow": {
    "xs": "0 0 0 1px rgb(31, 42, 68, 0.05)",
    "sm": "0 1px 2px 0 rgb(31, 42, 68, 0.05)",
    "default": "0 1px 3px 0 rgb(31, 42, 68, 0.1), 0 1px 2px 0 rgb(31, 42, 68, 0.06)",
    "md": "0 4px 6px -1px rgb(31, 42, 68, 0.1), 0 2px 4px -1px rgb(31, 42, 68, 0.06)",
    "lg": "0 10px 15px -3px rgb(31, 42, 68, 0.1), 0 4px 6px -2px rgb(31, 42, 68, 0.05)",
    "xl": "0 20px 25px -5px rgb(31, 42, 68, 0.1), 0 10px 10px -5px rgb(31, 42, 68, 0.04)",
    "2xl": "0 25px 50px -12px rgb(31, 42, 68, 0.25)",
    "inner": "inset 0 2px 4px 0 rgba(0, 0, 0, 0.06)",
    "outline": "0 0 0 3px rgba(66, 153, 225, 0.5)",
    "none": "none"
  },
  "fontFamily": {
    "title": [
      "Montserrat",
      "Lato",
      "system-ui",
      "-apple-system",
      "BlinkMacSystemFont",
      "Segoe UI",
      "Roboto",
      "Helvetica Neue",
      "Arial",
      "Noto Sans",
      "sans-serif"
    ],
    "sans": [
      "Lato",
      "system-ui",
      "-apple-system",
      "BlinkMacSystemFont",
      "Segoe UI",
      "Roboto",
      "Helvetica Neue",
      "Arial",
      "Noto Sans",
      "sans-serif"
    ],
    "serif": [
      "Times",
      "serif"
    ],
    "mono": [
      "monospace"
    ]
  },
  "fontWeight": {
    "normal": "400",
    "bold": "700"
  },
  "fontSize": {
    "xs": [
      "0.75rem",
      {
        "lineHeight": "1rem"
      }
    ],
    "sm": [
      "0.875rem",
      {
        "lineHeight": "1.25rem"
      }
    ],
    "base": [
      "1rem",
      {
        "lineHeight": "1.5rem"
      }
    ],
    "lg": [
      "1.125rem",
      {
        "lineHeight": "1.75rem"
      }
    ],
    "xl": [
      "1.25rem",
      {
        "lineHeight": "1.75rem"
      }
    ],
    "2xl": [
      "1.5rem",
      {
        "lineHeight": "2rem"
      }
    ],
    "3xl": [
      "1.875rem",
      {
        "lineHeight": "2.25rem"
      }
    ],
    "4xl": [
      "2.25rem",
      {
        "lineHeight": "2.5rem"
      }
    ],
    "5xl": [
      "3rem",
      {
        "lineHeight": "1"
      }
    ],
    "6xl": [
      "3.75rem",
      {
        "lineHeight": "1"
      }
    ],
    "7xl": [
      "4.5rem",
      {
        "lineHeight": "1"
      }
    ],
    "8xl": [
      "6rem",
      {
        "lineHeight": "1"
      }
    ],
    "9xl": [
      "8rem",
      {
        "lineHeight": "1"
      }
    ]
  }
}