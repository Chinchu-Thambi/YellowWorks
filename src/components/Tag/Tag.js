import React from 'react';
import PropTypes from 'prop-types';

import { BaseTag, RemoveTagButton } from './Styled';

/**
 * @readonly (optional) when present it doesn't renders the remove button.
 * @onRemove (required when readonly is false)
 */
const Tag = (props) => {
  const {
    children,
    readonly,
    onRemove,
    ...rest
  } = props;

  let removeButton = null;

  if (!readonly) {
    removeButton = (
      <RemoveTagButton
        type="button"
        title="Remove"
        aria-label="Remove"
        onClick={onRemove}
      >
        <svg viewBox="0 0 22 22" fill="none" xmlns="http://www.w3.org/2000/svg">
          <circle cx="11" cy="11" r="11" fill="#D2D4DA" />
          {/* eslint-disable-next-line max-len */}
          <path d="M11 6.90625V15.0927" stroke="white" strokeWidth="2" strokeMiterlimit="10" strokeLinecap="round" strokeLinejoin="round" />
          {/* eslint-disable-next-line max-len */}
          <path d="M7.11719 11L15.3032 11" stroke="white" strokeWidth="2" strokeMiterlimit="10" strokeLinecap="round" strokeLinejoin="round" />
        </svg>
      </RemoveTagButton>
    );
  }

  return (
    // eslint-disable-next-line react/jsx-props-no-spreading
    <BaseTag {...rest}>
      {children}
      {removeButton}
    </BaseTag>
  );
};

export default Tag;

Tag.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  readonly: PropTypes.bool,
  onRemove: PropTypes.func,
};

Tag.defaultProps = {
  readonly: false,
  children: [],
  onRemove: () => {},
};
