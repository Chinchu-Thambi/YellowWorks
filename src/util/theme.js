/* eslint-disable prefer-destructuring */
/* globals window */
import React from 'react';
import * as t from '../services/theme.json';

/** **********************************************************************************************
 * @deprecated
 *
 * For use with older rebass and styled components only.
 *
 * Use tailwind for styling new components. Refer to tailwind.config.js and services/theme.json.
 ************************************************************************************************ */


/**
 * SPEC REFERENCE AT https://github.com/system-ui/theme-specification
 */

/**
  * @deprecated
  * legacy - refer to canonical theme in services/theme.json
  */
const containerWidth = [
  1, // 100%
  '33.75rem', // 540px
  '43.75rem', // 720px
  '58.75rem', // 940px
  '73.75rem', // 1140px
  '90.00rem', // 1440px
];

/**
  * @deprecated
  * legacy - refer to canonical theme in services/theme.json
  */
const colors = {
  blue: '#07c', // palette.accent[4][0]
  brand: '#1F2A44', // palette.contrast[0]
  brandLight: '#343E55', // palette.contrast[1]
  primary: '#FBE122', // palette.brand[0]
  cerulean: '#00aded', // palette.accent[3][0]
  azure: '#345DA4', // palette.accent[4][0]
  black: '#000', // palette.contrast[0]
  dolphin: '#667', // palette.contrast[1]
  raven: '#797F8F', // palette.contrast[2]
  dustyGray: '#999', // palette.contrast[3]
  spunPearl: '#aab', // palette.contrast[3]
  silver: '#ccc', // palette.contrast[4]
  mischka: '#ddd', // palette.contrast[4]
  concrete: '#f2f2f2', // palette.contrast[5]
  pleinDeVie: '#BABABA', // palette.contrast[3]
  athensGray: '#f3f5f8', // palette.contrast[5]
  gallery: '#eee', // palette.base[3]
  lightGray: '#f6f6ff', // palette.contrast[5]
  white: '#fff', // palette.base[0],
  yellow: '#F9E01A',
};

const palette = {
  base: [
    t.colors.base['100'],
    t.colors.base['200'],
    t.colors.base['300'],
    t.colors.base['400'],
  ],
  brand: [ // The Optimist
    t.colors.brand['600'],
    t.colors.brand['500'],
    t.colors.brand['400'],
    t.colors.brand['300'],
    t.colors.brand['200'],
    t.colors.brand['100'],
  ],
  contrast: [ // Midnight
    t.colors.contrast['600'],
    t.colors.contrast['500'],
    t.colors.contrast['400'],
    t.colors.contrast['300'],
    t.colors.contrast['200'],
    t.colors.contrast['100'],
  ],
  accent: [
    [ // Crimson
      t.colors.crimson['600'],
      t.colors.crimson['500'],
      t.colors.crimson['400'],
      t.colors.crimson['300'],
      t.colors.crimson['200'],
      t.colors.crimson['100'],
    ],
    [ // Fern
      t.colors.fern['600'],
      t.colors.fern['500'],
      t.colors.fern['400'],
      t.colors.fern['300'],
      t.colors.fern['200'],
      t.colors.fern['100'],
    ],
    [ // Uplift
      t.colors.uplift['600'],
      t.colors.uplift['500'],
      t.colors.uplift['400'],
      t.colors.uplift['300'],
      t.colors.uplift['200'],
      t.colors.uplift['100'],
    ],
    [ // Pacific
      t.colors.pacific['600'],
      t.colors.pacific['500'],
      t.colors.pacific['400'],
      t.colors.pacific['300'],
      t.colors.pacific['200'],
      t.colors.pacific['100'],
    ],
    [ // Magnificent
      t.colors.magnificent['600'],
      t.colors.magnificent['500'],
      t.colors.magnificent['400'],
      t.colors.magnificent['300'],
      t.colors.magnificent['200'],
      t.colors.magnificent['100'],
    ],
    [ // Violet
      t.colors.violet['600'],
      t.colors.violet['500'],
      t.colors.violet['400'],
      t.colors.violet['300'],
      t.colors.violet['200'],
      t.colors.violet['100'],
    ],
  ],
};

/**
  * @deprecated
  * legacy - refer to canonical theme in services/theme.json
  */
// color aliases for palette.accent
palette.error = palette.accent[0];
palette.success = palette.accent[1];
palette.warn = palette.accent[2];
palette.navigation = palette.accent[3];
palette.support = palette.accent[4];

const theme = {
  breakpoints: [
    t.screens.sm,
    t.screens.md,
    t.screens.lg,
    t.screens.xl,
    t.screens['2xl'],
  ],
  containerWidth,
  fontSizes: [
    t.fontSize.sm[0],
    t.fontSize.xs[0],
    t.fontSize.base[0],
    t.fontSize['2xl'][0],
    t.fontSize['3xl'][0],
    t.fontSize['4xl'][0],
    t.fontSize['6xl'][0],
  ],
  // TODO replace usage of this...
  fontWeight: [
    t.fontWeight.normal,
    t.fontWeight.bold,
  ],
  // TODO ...with this (e.g. fontWeight="bold")
  fontWeights: { // PROPER SCHEMA
    body: t.fontWeight.normal,
    bold: t.fontWeight.bold,
  },
  zIndex: {
    dropdown: 1000,
    sticky: 1020,
    fixed: 1030,
    modalBackdrop: 1040,
    modal: 1050,
    popOver: 1060,
    tooltip: 1070,
  },
  colors,
  palette,
  space: [
    t.spacing['0'],
    t.spacing['1'],
    t.spacing['2'],
    t.spacing['3'],
    t.spacing['4'],
    t.spacing['5'],
    t.spacing['6'],
    t.spacing['7'],
    t.spacing['8'],
    t.spacing['9'],
    t.spacing['10'],
    t.spacing['11'],
    t.spacing['12'],
    t.spacing['13'],
    t.spacing['14'],
  ],
  fonts: {
    sans: t.fontFamily.sans[0],
    title: t.fontFamily.title[0],
  },
  shadows: {
    subtle: t.boxShadow.xs,
    small: t.boxShadow.sm,
    medium: t.boxShadow.md,
    large: t.boxShadow.lg,
  },
  // TODO should not have these buttons in the theme - delete later
  buttons: {
    primary: {
      color: 'black',
      backgroundColor: colors.primary,
      textTransform: 'uppercase',
      borderRadius: '40px',
      fontFamily: 'Montserrat',
      fontWeight: 500,
      fontSize: '0.875rem',
    },
    outline: {
      color: colors.blue,
      backgroundColor: 'transparent',
      boxShadow: 'inset 0 0 2px',
    },
  },
  radii: [
    t.borderRadius.none,
    t.borderRadius.default,
    t.borderRadius.md,
    t.borderRadius.lg,
  ],
};

/**
 * @deprecated
 * legacy - refer to canonical theme in services/theme.json and use tailwind to manage breakpoints
 */
const widthToBreakpoint = (width) => {
  if (width <= 576) return 'xs';
  if (width <= 768) return 'sm';
  if (width <= 992) return 'md';
  if (width <= 1200) return 'lg';
  if (width <= 1440) return 'xl';
  return 'xxl';
};

/**
 * @deprecated
 * legacy - refer to canonical theme in services/theme.json and use tailwind to manage breakpoints
 */
export const useBreakpoint = () => {
  if (typeof window === 'undefined') { return 'xs'; }
  // SSR for static pages will fail due to window useage - shift towards default window instead.
  // eslint-disable-next-line react-hooks/rules-of-hooks
  const [breakpoint, setBreakpoint] = React.useState(widthToBreakpoint(window.innerWidth));
  // eslint-disable-next-line react-hooks/rules-of-hooks
  React.useEffect(() => {
    const onResize = () => {
      setBreakpoint(widthToBreakpoint(window.innerWidth));
    };
    const resizeListener = window.addEventListener('resize', onResize);
    return () => {
      window.removeEventListener('resize', resizeListener);
    };
  });
  return breakpoint;
};

export default theme;

/**
 * @deprecated
 * legacy - refer to canonical theme in services/theme.json and use tailwind to manage breakpoints
 */
export const mediaBreakpoints = {
  xs: {
    minScreenWidth: 0,
    maxScreenWidth: 576 - 0.02,
  },
  sm: {
    minScreenWidth: theme.breakpoints[0],
    maxContainerWidth: containerWidth[1],
  },
  md: {
    minScreenWidth: theme.breakpoints[1],
    maxContainerWidth: containerWidth[2],
  },
  lg: {
    minScreenWidth: theme.breakpoints[2],
    maxContainerWidth: containerWidth[3],
  },
  xl: {
    minScreenWidth: theme.breakpoints[3],
    maxContainerWidth: containerWidth[4],
  },
  xxl: {
    minScreenWidth: theme.breakpoints[4],
    maxContainerWidth: containerWidth[5],
  },
};

/**
 * @deprecated
 * legacy - refer to canonical theme in services/theme.json and use tailwind to manage breakpoints
 */
export const mediaBreakpointUp = (size) => (
  mediaBreakpoints[size]
    ? `@media (min-width: ${mediaBreakpoints[size].minScreenWidth})`
    : ''
);
