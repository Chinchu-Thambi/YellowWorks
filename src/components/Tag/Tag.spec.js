/* globals describe, it, expect */
import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import theme from '../../util/theme';

import Tag from './Tag';

describe('Tag', () => {
  const modalChildren = (
    <div data-testid="child">I am a text</div>
  );

  it('renders without crashing', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Tag
          readonly
          onRemove={() => { }}
        >
          {modalChildren}
        </Tag>
      </ThemeProvider>,
    );

    expect(wrapper).toBeDefined();
    expect(wrapper).toBeTruthy();

    wrapper.unmount();
  });

  it('should have specific containers', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Tag
          onRemove={() => { }}
        >
          {modalChildren}
        </Tag>
      </ThemeProvider>,
    );

    expect(wrapper.find('button')).toHaveLength(1);
    expect(wrapper.find('button').prop('aria-label')).toBe('Remove');
    expect(wrapper.find('button').prop('title')).toBe('Remove');
    expect(wrapper.find('button svg')).toHaveLength(1);
    expect(wrapper.find('[data-testid="child"]').text()).toBe('I am a text');

    wrapper.unmount();
  });

  it('should not render remove button', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <Tag
          readonly
          onRemove={() => { }}
        >
          {modalChildren}
        </Tag>
      </ThemeProvider>,
    );

    expect(wrapper.find('button')).toHaveLength(0);

    wrapper.unmount();
  });

  it('should match snapshot', () => {
    const wrapper = mount(<ThemeProvider theme={theme}><Tag /></ThemeProvider>);
    expect(wrapper).toMatchSnapshot();

    wrapper.unmount();
  });
});
