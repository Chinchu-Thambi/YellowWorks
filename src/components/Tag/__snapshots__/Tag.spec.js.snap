// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should match snapshot 1`] = `
<Component
  theme={
    Object {
      "breakpoints": Array [
        "36em",
        "48em",
        "62em",
        "75em",
        "90em",
      ],
      "buttons": Object {
        "outline": Object {
          "backgroundColor": "transparent",
          "boxShadow": "inset 0 0 2px",
          "color": "#07c",
        },
        "primary": Object {
          "backgroundColor": "#FBE122",
          "borderRadius": "40px",
          "color": "black",
          "fontFamily": "Montserrat",
          "fontSize": "0.875rem",
          "fontWeight": 500,
          "textTransform": "uppercase",
        },
      },
      "colors": Object {
        "athensGray": "#f3f5f8",
        "azure": "#345DA4",
        "black": "#000",
        "blue": "#07c",
        "brand": "#1F2A44",
        "brandLight": "#343E55",
        "cerulean": "#00aded",
        "concrete": "#f2f2f2",
        "dolphin": "#667",
        "dustyGray": "#999",
        "gallery": "#eee",
        "lightGray": "#f6f6ff",
        "mischka": "#ddd",
        "pleinDeVie": "#BABABA",
        "primary": "#FBE122",
        "raven": "#797F8F",
        "silver": "#ccc",
        "spunPearl": "#aab",
        "white": "#fff",
        "yellow": "#F9E01A",
      },
      "containerWidth": Array [
        1,
        "33.75rem",
        "43.75rem",
        "58.75rem",
        "73.75rem",
        "90.00rem",
      ],
      "fontSizes": Array [
        "0.875rem",
        "0.75rem",
        "1rem",
        "1.5rem",
        "1.875rem",
        "2.25rem",
        "3.75rem",
      ],
      "fontWeight": Array [
        "400",
        "700",
      ],
      "fontWeights": Object {
        "body": "400",
        "bold": "700",
      },
      "fonts": Object {
        "sans": "Lato",
        "title": "Montserrat",
      },
      "palette": Object {
        "accent": Array [
          Array [
            "#D83F3D",
            "#E06564",
            "#E88C8B",
            "#EFB2B1",
            "#F7D9D8",
            "#FDF5F5",
          ],
          Array [
            "#64B056",
            "#83C078",
            "#A2D09A",
            "#C1DFBB",
            "#E0EFDD",
            "#F7FBF6",
          ],
          Array [
            "#ED7047",
            "#F18D6C",
            "#F4A991",
            "#F8C6B5",
            "#FBE2DA",
            "#FEF8F6",
          ],
          Array [
            "#6BBEE7",
            "#89CBEC",
            "#A6D8F1",
            "#C4E5F5",
            "#E1F2FA",
            "#F7FBFE",
          ],
          Array [
            "#6D3A96",
            "#8A61AB",
            "#A789C0",
            "#C5B0D5",
            "#E2D8EA",
            "#F7F5FA",
          ],
          Array [
            "#61679E",
            "#8185B1",
            "#A0A4C5",
            "#C0C2D8",
            "#DFE1EC",
            "#F7F7FA",
          ],
        ],
        "base": Array [
          "#FFFFFF",
          "#F4F4F4",
          "#F2F2F2",
          "#E7E7E7",
        ],
        "brand": Array [
          "#FBE122",
          "#FCE74E",
          "#FDED7A",
          "#FDF3A7",
          "#FEF9D3",
          "#FFFDF4",
        ],
        "contrast": Array [
          "#1F2A44",
          "#4C5569",
          "#797F8F",
          "#A5AAB4",
          "#D2D4DA",
          "#F3F5F8",
        ],
        "error": Array [
          "#D83F3D",
          "#E06564",
          "#E88C8B",
          "#EFB2B1",
          "#F7D9D8",
          "#FDF5F5",
        ],
        "navigation": Array [
          "#6BBEE7",
          "#89CBEC",
          "#A6D8F1",
          "#C4E5F5",
          "#E1F2FA",
          "#F7FBFE",
        ],
        "success": Array [
          "#64B056",
          "#83C078",
          "#A2D09A",
          "#C1DFBB",
          "#E0EFDD",
          "#F7FBF6",
        ],
        "support": Array [
          "#6D3A96",
          "#8A61AB",
          "#A789C0",
          "#C5B0D5",
          "#E2D8EA",
          "#F7F5FA",
        ],
        "warn": Array [
          "#ED7047",
          "#F18D6C",
          "#F4A991",
          "#F8C6B5",
          "#FBE2DA",
          "#FEF8F6",
        ],
      },
      "radii": Array [
        "0",
        "0.15625rem",
        "0.3125rem",
        "0.625rem",
      ],
      "shadows": Object {
        "large": "0 10px 15px -3px rgb(31, 42, 68, 0.1), 0 4px 6px -2px rgb(31, 42, 68, 0.05)",
        "medium": "0 4px 6px -1px rgb(31, 42, 68, 0.1), 0 2px 4px -1px rgb(31, 42, 68, 0.06)",
        "small": "0 1px 2px 0 rgb(31, 42, 68, 0.05)",
        "subtle": "0 0 0 1px rgb(31, 42, 68, 0.05)",
      },
      "space": Array [
        "0rem",
        "0.3125rem",
        "0.625rem",
        "1.25rem",
        "2rem",
        "3rem",
        "4rem",
        "5rem",
        "6rem",
        "7rem",
        "8rem",
        "10rem",
        "12rem",
        "14rem",
        "16rem",
      ],
      "zIndex": Object {
        "dropdown": 1000,
        "fixed": 1030,
        "modal": 1050,
        "modalBackdrop": 1040,
        "popOver": 1060,
        "sticky": 1020,
        "tooltip": 1070,
      },
    }
  }
>
  <Tag
    onRemove={[Function]}
    readonly={false}
  >
    <styled.span>
      <span
        className="sc-bdnylx hDeDJa"
      >
        <styled.button
          aria-label="Remove"
          onClick={[Function]}
          title="Remove"
          type="button"
        >
          <button
            aria-label="Remove"
            className="sc-gtssRu eGpIiL"
            onClick={[Function]}
            title="Remove"
            type="button"
          >
            <svg
              fill="none"
              viewBox="0 0 22 22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="11"
                cy="11"
                fill="#D2D4DA"
                r="11"
              />
              <path
                d="M11 6.90625V15.0927"
                stroke="white"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeMiterlimit="10"
                strokeWidth="2"
              />
              <path
                d="M7.11719 11L15.3032 11"
                stroke="white"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeMiterlimit="10"
                strokeWidth="2"
              />
            </svg>
          </button>
        </styled.button>
      </span>
    </styled.span>
  </Tag>
</Component>
`;
